### ISSUE TITLE
No tests found on net461 project with 17.4.0 - Improve the output

### ISSUE THREAD

**Issue by ermshiperete** [microsoft/vstest#4187]
## Description

When building a net461 project that references Microsoft.NET.Test.Sdk 17.4.0 and then running `dotnet test`, no test are found.
Explicitly specifying the test assembly works.

This first looked like a problem with the NUnit3TestAdapter 4.3.x, but it turns out that it's caused by Microsoft.NET.Test.Sdk 17.4.0.

See https://github.com/nunit/nunit3-vs-adapter/issues/1040

## Steps to reproduce

- Clone test project https://gist.github.com/ermshiperete/820a93a320e300a52ab569e00329a397
- run `dotnet test`

## Expected behavior

Tests discovered and run, or at least a warning or error that net461 isn't supported in 17.4.0

## Actual behavior

Project builds without errors or warnings, but no tests are found

# Environment

Ubuntu 20.04 LTS
dotnet 6.0.403

**Comment by Evangelink** [comment]
I confirm the behavior you are seeing. Between `17.3.2` and `17.4` we have aligned the version supported following Microsoft .NET Framework lifecycle (see https://learn.microsoft.com/lifecycle/products/microsoft-net-framework) which I suspect is causing this issue.

**Comment by MarcoRossignoli** [comment]
I'm going to close because behavior is by-design, feel free to reopen if needed.

**Comment by MarcoRossignoli** [comment]
Thanks @ermshiperete for the suggestion I'll reopen for now and I'll discuss with the team.

**Comment by pasisavolainen** [comment]
This is a bit weirder still.
For us with
`<TargetFramework>net461</TargetFramework>`
will discover+run tests:
```xml
<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.5.0" />
<PackageReference Include="MSTest.TestAdapter" Version="2.2.8" />
<PackageReference Include="MSTest.TestFramework" Version="2.2.8" />
```

this will discover and show hierarchical/datatests but will not run them:
```xml
<PackageReference Include="Microsoft.NET.Test.Sdk" Version="17.5.0" />
<PackageReference Include="MSTest.TestAdapter" Version="3.0.2" />
<PackageReference Include="MSTest.TestFramework" Version="3.0.2" />
```

**Comment by MarcoRossignoli** [comment]
> this will discover and show hierarchical/datatests but will not run them:

Can you be more specific? Do you mean it's correctly discovering inside Visual Studio Test Explorer but it's not running the tests? If so can you try to get the logs following this guide(pls note the disclaimer on the sensitive information)? https://github.com/microsoft/vstest/blob/main/docs/troubleshooting.md#visual-studio

**Comment by pasisavolainen** [comment]
> Do you mean it's correctly discovering inside Visual Studio Test Explorer but it's not running the tests?

Yes, Test Explorer shows the tests (after `Clean Solution` + `Rebuild Solution`). They're obviously with the 'not run` state:
[2023_03_07_17_27_04_15004.zip](https://github.com/microsoft/vstest/files/10911111/2023_03_07_17_27_04_15004.zip)

![image](https://user-images.githubusercontent.com/6162135/223463032-3ae1b1ae-0c0b-4775-8011-237aa7b7f80f.png)

uh. waited a bit and they ran.
- changed back to "2.2.8" testadapter/testframework and tests don't run
- rebuild, doesn't run
- changed test tracing to 'error', doesn't run
- clicked multiple times 'Run All Tests in View', 0 tests run.
- rebuild, now they run.

Attached logs of:
- fresh VS start with 2.2.8 situation, TestsLog=Trace
- change testadapter+testframework to 3.0.2
- rebuild
- 'Run all tests in view' -> doesn't run (situation as in screenshot above)
